src/components/Countdown/Countdown.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import CountdownUnit from './CountdownUnit';
+import PropTypes from 'prop-types';
+import styles from './CountdownContainer.module.scss';
+
+const Countdown = ({ countdown }) => {
+  return (
+    <div className={styles.countdown}>
+      <CountdownUnit value={countdown.days} label="Days" />
+      <h2>:</h2>
```

Comment:
Thanks for making me aware of the use of `var(--border-color)`. What do you use for the headings and text that switches from dark to light mode? I have been trying to find it but I am new to both the codebase and scss haha

Reply:
That's a really great question! There are a few moving pieces to this equation.

tl;dr; 

We have classes that get added to the `body` tag, and we use [CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) to switch the primary text color depending on which "mode" the site is in (dark or light). You can see the common global variables here: https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/GlobalStyles.js#L104-L146

### Deep dive

We are using a Gatsby plugin called [`gatsby-plugin-use-dark-mode`](https://www.gatsbyjs.com/plugins/gatsby-plugin-use-dark-mode/) which, in turn uses a library called [`use-dark-mode`](https://github.com/donavon/use-dark-mode#readme). Essentially these libraries are responsible for toggling the class on the `body` tag between `light-mode` and `dark-mode` (Try it! If you inspect the body tag and click the dark mode toggle, you can see this in action 🙂 ). 

We use the `gatsby-plugin-use-dark-mode` plugin in our [`gatsby-config.js`](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/gatsby-config.js#L7) for the theme and use the `use-dark-mode` hook in the [`DarkModeToggle`](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/DarkModeToggle.js) component (which is the element you click on in the header to toggle the mode!). 

We take advantage of the `light-mode` or `dark-mode` class that gets toggled on the `body` to swap out the values of the CSS variables. If you look at our [global styles](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/GlobalStyles.js#L104-L146), you'll notice that we can use a CSS variable as the value of another CSS variable:

```css
--primary-text-color: var(--color-neutrals-700);
```

This is really powerful, because it allows us to create abstractions in our CSS. Combined with our [color](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/GlobalStyles.js#L5-L102) definitions, we can then target the mode and set the appropriate text color accordingly:

```css
.light-mode {
  --primary-text-color: var(--color-neutrals-700);
}

.dark-mode {
  --primary-text-color: var(--color-dark-700);
}
```
(source: [light mode](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/GlobalStyles.js#L107), [dark mode](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/GlobalStyles.js#L129))

Whats super cool about that is that now I can use the `--primary-text-color` variable anywhere I want "dark mode aware" text that matches the prominent text color used throughout the site. In fact, that's exactly what we do for the `body` selector by setting the [`color` property](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/GlobalStyles.js#L173) to `--primary-text-color`. I find that to be pretty cool 🎉 .

For headings, we do something very similar. As you've noticed on the site, our headings are much brighter than the primary text color. Here we've defined a `--heading-text-color` for [light mode](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/GlobalStyles.js#L122) and [dark mode](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/GlobalStyles.js#L144), which we then use to set the [heading colors](https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/GlobalStyles.js#L203).

Whew, that was a lot 😄 . Hopefully that helps you get a little better understanding of some of the mechanics we've employed to pull off our dark mode theme and where to find them in the code base!

Reply:
> What do you use for the headings and text that switches from dark to light mode? 

Because it probably wasn't super obvious in that last comment you can use the `var(--heading-text-color)` variable to match the heading text color styles, which is dark mode aware.